tests/test_populate_catalog.py

```diff
@@ -652,3 +655,24 @@ def should_add_link_to_root_catalog_in_series(
                 expected_root_catalog_links
                 == load(smart_open.open(root_url, mode="rb"))[STAC_LINKS_KEY]
             )
+
+
+@mark.infrastructure
+@patch("geostore.populate_catalog.task.read_file")
+def should_log_error_message_when_an_exception_is_raised_trying_to_update_catalog(
+    pystac_read_file_mock: MagicMock,
+) -> None:
+    exception_class = any_exception_class()
+    error_message = any_error_message()
+    error = exception_class(error_message)
+    expected_message = f"{LOG_MESSAGE_LAMBDA_FAILURE}: Unable to populate catalog due to “{error}”"
+
+    pystac_read_file_mock.side_effect = error
+
+    with patch("geostore.populate_catalog.task.LOGGER.warning") as logger_mock:
+        with raises(exception_class):
```

Comment:
[`raises` can be part of the previous `with`](https://github.com/linz/geostore/pull/2098#discussion_r987378850).